python/lsst/pipe/drivers/constructCalibs.py

```diff
@@ -375,20 +375,26 @@ def run(self, expRefList, butler, calibId):
         @param butler      Data butler
         @param calibId   Identifier dict for calib
         """
+        for expRef in expRefList:
+            self.addMissingKeys(expRef.dataId, butler, self.config.ccdKeys, 'raw')
+
         outputId = self.getOutputId(expRefList, calibId)
         ccdIdLists = getCcdIdListFromExposures(
```

Comment:
How about doing something like:

    expIdList = [[ccdRef.dataId for ccdRef in expRef.subItems("sensor")] for expRef in expRefList]
    ccdIdLists = getCcdIdListFromExposures(expIdList, ccdKeys=self.config.ccdKeys)

and redefining `getCcdIdListFromExposures` to:

    def getCcdIdListFromExposures(expIdList, ccdKeys=["ccd"]):

That way, you'll get the important information in there once and for all --- it won't have to be repeated in `scatterCombine`.